governor/src/state/in_memory.rs

```diff
@@ -44,6 +44,33 @@ impl InMemoryState {
         decision.map(|(result, _)| result)
     }
 
+    pub(crate) fn measure_and_peek_one<T, F, E>(&self, mut f: F) -> Result<T, E>
+    where
+        F: FnMut(Option<Nanos>) -> Result<(T, Nanos), E>,
+    {
+        let mut prev = self.0.load(Ordering::Acquire);
+        let original_prev = prev;
+        let mut decision = f(NonZeroU64::new(prev).map(|n| n.get().into()));
+        while let Ok((result, new_data)) = decision {
+            match self.0.compare_exchange_weak(
+                prev,
+                new_data.into(),
+                Ordering::Release,
+                Ordering::Relaxed,
+            ) {
+                Ok(_) => {
+                    self.0.store(original_prev, Ordering::Release);
+                    return Ok(result);
+                }
+                Err(next_prev) => prev = next_prev,
+            }
+            decision = f(NonZeroU64::new(prev).map(|n| n.get().into()));
```

Comment:
Looks like codecov thinks L65-L67 aren't exercised in tests; would be great if there was a way to test those too.

Reply:
I think I got this one now covered. Interestingly, now that `codecov` is succeeding, I cannot find the individual cover report any more. Should there be anything left, please let me know.